dill/_objects.py

```diff
@@ -468,14 +471,23 @@ class _Struct(ctypes.Structure):
 x['HashType'] = hashlib.md5()
 x['HMACType'] = hmac.new(_in)
 # generic operating system services (CH 15)
-#x['CursesWindowType'] = _curwin = curses.initscr() #FIXME: messes up tty
-#x['CursesTextPadType'] = textpad.Textbox(_curwin)
-#x['CursesPanelType'] = panel.new_panel(_curwin)
+try:
+    import curses
+    from curses import textpad, panel
+    #x['CursesWindowType'] = _curwin = curses.initscr() #FIXME: messes up tty
+    #x['CursesTextPadType'] = textpad.Textbox(_curwin)
+    #x['CursesPanelType'] = panel.new_panel(_curwin)
+except ImportError:
+    pass
+    
 if HAS_CTYPES:
     x['CCharPType'] = ctypes.c_char_p()
     x['CWCharPType'] = ctypes.c_wchar_p()
     x['CVoidPType'] = ctypes.c_void_p()
-    x['CDLLType'] = _cdll = ctypes.CDLL(None)
+    if sys.platform == 'win32':
```

Comment:
This was just an example. The previous one https://www.python.org/dev/peps/pep-0345/#environment-markers has the same syntax. This the syntax used and implemented by pip and setuptools.